lib/worker-type.js

```diff
@@ -339,6 +339,59 @@ WorkerType.prototype.testLaunchSpecs = function () {
 };
 
 /**
+ * Return a list of every scope for every region and instance type combination.
+ * Currently, scope [1,2] is considered unique compared to [2,1].  it would be
+ * nice if we could convert the array of arrays into a set of sets for
+ * comparisons
+ */
+WorkerType.scopes = function (worker) {
+  // Generate the complete list of scopes;
+  let scopes = [];
+
+  // Take an object with a .scopes property and list add all of its (mostly)
+  // unique scopes to the list of scopes to return
+  var y = function(x) {
+    for (let scope of x.scopes) {
+      if (typeof scope === 'string') {
+        scope = [scope];
+      }
+      let hasThisScope = false;
+      for (let knownScope of scopes) {
+        if (lodash.isEqual(scope, knownScope)) {
+          hasThisScope = true;
+          break;
+        }
+      }
+      if (!hasThisScope) {
+        scopes.push(scope);
+      }
+    }
+  }
+
+  // Global scopes
+  y(worker);
+
+  // Region specific scopes
+  for (let region of worker.regions) {
+    y(region);
+  }
+
+  // Instance Type specific scopes
+  for (let instanceType of worker.instanceTypes) {
+    y(instanceType);
+  }
+
+  return scopes; 
+};
```

Comment:
This seems way to complicated... and wrong.
Just do:
```js
return _.union(
  worker.scopes,
  _.flatten(worker.instanceTypes.map(t => t.scopes)),
  _.flatten(worker.regions.map(r => r.scopes))
]));
```
I assume you're code always creates the `scopes` property as an array, even if it's empty.
If you don't do that... then I recommend that you start doing so, and do a version upgrade then you can fix the old instances in the `migrate` function.

Sure you could add `|| []` everywhere but that is bad... Defaults should exists in JSON schemas and if not there, then when objects are created. But I'm pretty sure you can settle with just having them in your JSON schemas (for the input).